Documentation/check-cmdref.sh

```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+script_dir="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+cmdref_dir="${script_dir}/cmdref"
+
+if ! git diff --quiet -- "${cmdref_dir}" ; then
+    git --no-pager diff "${cmdref_dir}"
+    echo "HINT: to fix this, run 'make -C Documetation update-cmdref ; git commit Documetation/cmdref --message \"docs: Update cmdref\"'"
```

Comment:
```suggestion
    echo "HINT: to fix this, run 'make -C Documentation update-cmdref ; git commit Documentation/cmdref --message \"docs: Update cmdref\"'"
```

Reply:
This check currently fails for me, when running `make -j$(nproc)` on a clean clone. It reports all cmdref files as missing:
```
rm -rvf cmdref/cilium*.md
docker container run --rm --workdir /src/Documentation --volume /home/paul/cilium/Documentation/..:/src --user "1000:1000" cilium/docs-builder ./check-cmdref.sh
removed 'cmdref/cilium-agent.md'
removed 'cmdref/cilium-bugtool.md'
[...]
removed 'cmdref/cilium_version.md'
docker container run --rm --workdir /src/Documentation --volume /home/paul/cilium/Documentation/..:/src --user "1000:1000" cilium/docs-builder ./update-cmdref.sh
diff --git a/Documentation/cmdref/cilium-agent.md b/Documentation/cmdref/cilium-agent.md
deleted file mode 100644
index 2729719e1..000000000
--- a/Documentation/cmdref/cilium-agent.md
+++ /dev/null
@@ -1,174 +0,0 @@
-<!-- This file was autogenerated via cilium-agent --cmdref, do not edit manually-->
-
[...]
diff --git a/Documentation/cmdref/cilium-bugtool.md b/Documentation/cmdref/cilium-bugtool.md
deleted file mode 100644
index acd702818..000000000
--- a/Documentation/cmdref/cilium-bugtool.md
+++ /dev/null
[...]
```

Reply:
@pchaigno I am not sure how to fix this to be honest, I am not yet equipped to run the entire build locally, and overall the Makefile is pretty complex, this could be to do with anything really. If you posted the full output, I can take a look and see what it maybe. I am not sure if we really must support `-j`, in my experience it break quite often, unless it's specifically be take care of.
Also, `-j` is most helpful when you have to compile a lot of `.c` files in `.o` files and you are invoking the compiler for each one of them, which is something you don't need to worry about with `go build`  as it runs as many gorouties as it sees fit. It's even less useful when you are invoking `docker run`. Basically you need to get dependencies really straight for `-j` to work well, we declare some dependencies, but it's also done a little naively right now. I also have serious doubts about using make dependencies with `go build`, especially since build cache was improved recently.

Reply:
Aside from what I said about - does `make -j$(nproc)` work on master? If it did, I will have to fix it here or follow-up with another PR.

Reply:
Yes, `make -j12` works on master and saves about 32 seconds on a 53 seconds build compared to `make`. I haven't looked into what is actually impacted (maybe BPF compilation?).

I'm sending you the full output of `make -j12 &> build-output.log` via Slack (GitHub upload seems to be temporarily failing).

Reply:
> saves about 32 seconds on a 53 seconds build

That's not for a clean build, right?

Reply:
I think it is. At least, I ran `make clean` and `git clean -fdx` prior to those test builds.

Reply:
With this pull request, running `make` takes 10 more seconds on the first run and 10 less seconds once Docker images are prepared. Same impact on `make -j12`. I'm down to a <10s build time! :tada:

Reply:
That's a nice side-effect indeed,  but the main goal was to actually make doc issues fail faster with GitHub Actions instead of Jenkins.

Reply:
The `make -j` issue should now be resolved per Slack discussion.